src/ui/components/abstract/LayoutFrame.mjs

```diff
@@ -0,0 +1,30 @@
+class LayoutFrame {
+  constructor(ui) {
+    this.ui = ui;
+
+    // TODO: Risk of flags prop clashing with other classes in hierarchy?
```

Comment:
This will be a regular issue, I think. If I understand C++ correctly, subclasses can have members with the same name as superclasses, and will prefer the most specific member based on the type used to access it.

I suspect we'll have to uniquely name the properties to avoid this problem in JS. `super` seems to be restricted to `prototype`s only.

Reply:
Yeah 😢 👍